### DISCUSSION TITLE
Failed to download Firedrake on Mac

### DISCUSSION THREAD

**Discussion by jmm1828** [firedrakeproject/firedrake#3005] [Firedrake support]
After trying to install firedrake with

`python3 firedrake-install`

I ran into the following error:

> No dependencies found. Skipping.
> Installing pip dependencies for FInAT
> Installing pip dependencies for tsfc
> Installing pip dependencies for PyOP2
> Installing pip dependencies for pyadjoint
> No dependencies found. Skipping.
> Installing pip dependencies for pytest-mpi
> No dependencies found. Skipping.
> Installing pip dependencies for firedrake
> Traceback (most recent call last):
>   File "/Users/javiermagide/Documents/Research_Project_Iain_Smears/firedrake-install", line 1907, in <module>
>     pip_requirements(p, compiler_env)
>   File "/Users/javiermagide/Documents/Research_Project_Iain_Smears/firedrake-install", line 1014, in pip_requirements
>     run_pip_install_wrap(reqs, parallel_compiler_env)
>   File "/Users/javiermagide/Documents/Research_Project_Iain_Smears/firedrake-install", line 1007, in run_pip_install_wrap
>     run_pip_install([req])
>   File "/Users/javiermagide/Documents/Research_Project_Iain_Smears/firedrake-install", line 979, in run_pip_install
>     check_call(pipinstall + pipargs)
>   File "/Users/javiermagide/Documents/Research_Project_Iain_Smears/firedrake-install", line 666, in check_call
>     log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
>               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 466, in check_output
>     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/Cellar/python@3.11/3.11.4/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 571, in run
>     raise CalledProcessError(retcode, process.args,
> subprocess.CalledProcessError: Command '['/Users/javiermagide/Documents/Research_Project_Iain_Smears/firedrake/bin/python', '-m', 'pip', 'install', '--no-build-isolation', '--no-binary', 'mpi4py,randomgen,islpy,numpy', '-vvv', 'matplotlib']' returned non-zero exit status 2.
> 

Bellow are the files firedrake-install.log and configure.log. Any help will be much appreciated.
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/11860569/firedrake-install.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/11860633/configure.log)

**Comment by connorjward** [comment]
One of the last lines in `firedrake-install.log` reads

```
pip._vendor.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.
```

It looks like your network dropped part way through the installation. Please could you remove the `firedrake` virtual environment and try again?

**Reply by jmm1828** [reply]
I will try that now, thank you.
